### PR TITLE
Add regression tests for two previously fixed bugs.

### PR DESCRIPTION
**Pull request by erozenfeld**
One of the repros is in il rather than C# because the
repro was sensitive to what different versions of C#
compiler generated.

### CODE REVIEW

**Comment by erozenfeld** [comment]
@cmckinsey @briansull PTAL 
/cc @dotnet/jit-contrib

**Comment by briansull** [comment]
LGTM

**Comment by cmckinsey** [comment]
LGTM. Thanks for adding these Eugene!

File: tests/src/JIT/Regression/JitBlue/DevDiv_150586/DevDiv_150586.il [hunk 1]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
```
Comment (sivarv) [inline tests/src/JIT/Regression/JitBlue/DevDiv_150586/DevDiv_150586.il]: Please add a couple of lines to indicate what this test is meant to test.

**Comment by sivarv** [comment]
Looks good.

File: tests/src/JIT/Regression/JitBlue/DevDiv_150586/DevDiv_150586.il (hunk 1 above)
Comment (erozenfeld) [inline tests/src/JIT/Regression/JitBlue/DevDiv_150586/DevDiv_150586.il]: Added a comment.

**Pull request merged**